cmd/k8s-operator/connector.go

```diff
@@ -272,19 +301,37 @@ func (a *ConnectorReconciler) validate(cn *tsapi.Connector) error {
 }
 
 func validateSubnetRouter(sb *tsapi.SubnetRouter) error {
-	if len(sb.AdvertiseRoutes) < 1 {
+	if len(sb.AdvertiseRoutes) == 0 {
 		return errors.New("invalid subnet router spec: no routes defined")
 	}
+	return validateRoutes(sb.AdvertiseRoutes)
+}
+
+func validateAppConnector(ac *tsapi.AppConnector) error {
+	if len(ac.Routes) == 0 {
+		return nil
+	}
+	return validateRoutes(ac.Routes)
+}
+
+func validateRoutes(routes tsapi.Routes) error {
 	var err error
```

Comment:
This function looks like it was written before some of Go's more recent `errors` library additions. I think it could be better expressed now as building up a slice of errors `var errs []error` and then return `errors.Join(errs)` at the end.